src/app/icd/ICDData.h

```diff
@@ -0,0 +1,127 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <platform/CHIPDeviceConfig.h>
+#include <system/SystemClock.h>
+
+#ifndef ICD_ENFORCE_SIT_SLOW_POLL_LIMIT
+// Set to 1 to enforce SIT Slow Polling Max value to 15seconds (spec 9.16.1.5)
+#define ICD_ENFORCE_SIT_SLOW_POLL_LIMIT 0
+#endif
+
+namespace chip {
+
+namespace app {
+// Forward declaration of ICDManager to allow it to be friend with ICDData
+class ICDManager;
+} // namespace app
+
+/**
+ * @brief ICDData manages and stores ICD related configurations for the ICDManager.
+ *        Goal of the class is to expose ICD information to all consummers without creating circular dependencies
+ *        since the ICDManager is appart of the App layer
+ *
+ *        Anyone can read the ICD configurations but only the ICDManager can changes those configurations.
+ */
+class ICDData
```

Comment:
Is this really ICDData (sounds very generic; I initially thought this was data on what things are registered) or more like ICDConfiguration?

Reply:
Renamed the class `ICDConfigurationData`. I would prefer keeping the `*Data` to indicate that the class just manages data and does not have any business logic.